### PR TITLE
update doc for DataDesc

### PR DESCRIPTION
**Pull request by nswamy**
Reformat IRHeader and Update Doc for DataDesc
@madjam, @piiswrong please review and merge.

### CODE REVIEW

File: python/mxnet/io.py [hunk 1]
```diff
@@ -17,7 +17,16 @@
...
 class DataDesc(namedtuple('DataDesc', ['name', 'shape'])):
-    """Data description
+    """DataDesc is used to store name, shape, type and layout
+        information of the data or the label.
+
+       The `layout` describes how the axes in `shape` should be interpreted,
+       for example for image data setting `layout=NCHW` indicates
+        that the first axis is number of examples in the batch(N),
+        C is number of channels, H is the height and W is the width of the image.
+       for sequential data the `layout` could be set to ``TNC`` where
```
Comment (piiswrong) [inline python/mxnet/io.py]: The default is NTC so let's say that here

File: python/mxnet/recordio.py [hunk 2]
```diff
@@ -274,16 +274,17 @@ def write_idx(self, idx, buf):
...
 ----------
-flag : int
-    Available for convenience, can be set arbitrarily.
-label : float or an array of float
-    Typically used to store label(s) for a record.
-id: int
-    Usually a unique id representing record.
-id2: int
-    Higher order bits of the unique id, should be set to 0 (in most cases).
+    flag : int
```
Comment (piiswrong) [inline python/mxnet/recordio.py]: This shouldn't be indented.

Comment (nswamy) [inline python/mxnet/recordio.py]: formatting for IRHeader is lost, see http://mxnet.io/api/python/io.html#mxnet.recordio.IRHeader.
How do we fix the formatting ?

Comment (piiswrong) [inline python/mxnet/recordio.py]: are you sure indenting these lines would fix it?

Comment (nswamy) [inline python/mxnet/recordio.py]: I can't get the build to work on my machine to verify how it renders. I followed the formatting of other APIs. 
It is already broken, can we get this in and see if it helps.

**Pull request merged**